crates/ty_ide/src/semantic_tokens.rs

```diff
@@ -2230,6 +2232,49 @@ class MyClass:
         "###);
     }
 
+    #[test]
+    fn test_constant_variations() {
+        let test = SemanticTokenTest::new(
+            r#"
+A = 1
+AB = 1
+ABC = 1
+A1 = 1
+AB1 = 1
+ABC1 = 1
+A_B = 1
+A1_B = 1
+A_B1 = 1
+A_1 = 1
+"#,
+        );
+
+        let tokens = test.highlight_file();
+
+        assert_snapshot!(test.to_snapshot(&tokens), @r#"
+        "A" @ 1..2: Variable [definition]
+        "1" @ 5..6: Number
+        "AB" @ 7..9: Variable [definition, readonly]
```

Comment:
...we classify these as readonly? That feels like a pyrightism that crept into the code somewhere. We might want to reconsider that 🤔

Pyright implements a heuristic where SCREAMING_SNAKE_CASE constants are always treated as implicitly read-only. But in terms of the semantics _we_ give variables, we currently only treat variables as read-only if they are explicitly annotated as `Final` (as per the spec).

If this classification only affects the way these variables are syntax-highlighted by the editor, though, then maybe it's fine